### PR TITLE
Sema: improvements to error messages related to the handling of (error) values

### PR DESCRIPTION
**Pull request by wooster0**
The main focus here was that the compiler suggested using `try`, `catch`, or `if` for handling errors in the form of error sets, but none of those work on error sets. This PR fixes that and a few other things.

### CODE REVIEW

File: src/Sema.zig [hunk 1]
```diff
@@ -26419,6 +26432,9 @@ fn fieldCallBind(
             const decl = mod.declPtr(decl_idx);
             try mod.errNoteNonLazy(decl.srcLoc(mod), msg, "'{}' is not a member function", .{field_name.fmt(ip)});
         }
+        if (concrete_ty.zigTypeTag(mod) == .ErrorUnion) {
```
Comment (Vexu) [inline src/Sema.zig:26435]: The error message doesn't include a type so either this should be removed or the message should be changed. Seeing that this note does not appear to be tested anywhere I'd go with the former.

**Review by Vexu — APPROVED** [review]

File: src/Sema.zig (hunk 1 above)
Comment (wooster0) [reply src/Sema.zig:26435]: https://github.com/ziglang/zig/pull/16461/commits/6f36e99899c6451bd3c1b8b8ef442247c48c529e

The test is test/cases/compile_errors/method_call_on_error_union.zig

Comment (wooster0) [reply src/Sema.zig:26435]: If you do what the error message I added says and do for example `(x catch unreachable).a()` the code compiles. That was the purpose of the commit.

Comment (Vexu) [reply src/Sema.zig]: My bad, seems like I mistook the above error note as the actual error message.

**Comment by andrewrk** [comment]
I'm sorry, I didn't review this in time, and now it has bitrotted. Furthermore, so many pull requests have stacked up that I can't keep up and I am therefore declaring Pull Request Bankruptcy and closing old PRs that now have conflicts with master branch.

If you want to reroll, you are by all means welcome to revisit this changeset with respect to the current state of master branch, and there's a decent chance your patch will be reviewed the second time around.

Either way, I'm closing this now, otherwise the PR queue will continue to grow indefinitely.

**Pull request closed without merging**